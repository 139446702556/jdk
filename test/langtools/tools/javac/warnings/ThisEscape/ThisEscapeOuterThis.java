/*
 * @test /nodynamiccopyright/
 * @bug 8194743
 * @compile/ref=ThisEscapeOuterThis.out -Xlint:this-escape -XDrawDiagnostics ThisEscapeOuterThis.java
 * @summary Verify 'this' escape detection handles leaks via outer 'this'
 */

public class ThisEscapeOuterThis {

    public ThisEscapeOuterThis() {
        new InnerClass();
    }

    public void mightLeak() {
    }

    public class InnerClass {

        InnerClass() {
            ThisEscapeOuterThis.this.mightLeak();
        }
    }
}
