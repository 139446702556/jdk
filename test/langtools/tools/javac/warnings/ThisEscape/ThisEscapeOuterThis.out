ThisEscapeOuterThis.java:20:47: compiler.warn.possible.this.escape
ThisEscapeOuterThis.java:11:9: compiler.warn.possible.this.escape.location
2 warnings
